src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperScopeManager.cs

```diff
@@ -124,5 +127,56 @@ public TagHelperExecutionContext End()
 
             return null;
         }
+
+        private class ExecutionContextPool
+        {
+            private readonly List<TagHelperExecutionContext> _executionContexts;
+            private int _nextIndex;
+
+            public ExecutionContextPool()
+            {
+                _executionContexts = new List<TagHelperExecutionContext>();
+            }
+
+            public TagHelperExecutionContext Create(
```

Comment:
Leave `Create` for methods that always create something. This is more like `Rent()` (which lines up with some other pools) or `GetContext()` (which lines up with a few other pools / buffers).